File: frontend/src/api/models/UnitUpdateRequestDTO.ts
import type { AddressDTO } from './AddressDTO.ts'

export type UnitUpdateRequestDTO = {
  /**
   * @type string, uuid
   */
  id: string
  /**
   * @type string
   */
  block: string
  /**
   * @type string
   */
  lot: string
  /**
   * @type object
   */
  address: AddressDTO
}

File: backend/src/main/java/br/com/visitsafe/dto/unit/UnitUpdateRequestDTO.java
package br.com.visitsafe.dto.unit;

import br.com.visitsafe.dto.AddressDTO;
import jakarta.validation.Valid;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import lombok.AllArgsConstructor;
import lombok.Data;

import java.util.UUID;

@Data
@AllArgsConstructor
public class UnitUpdateRequestDTO {

    @NotNull
    private UUID id;

    @NotBlank
    private String block;

    @NotBlank
    private String lot;

    @NotNull
    @Valid
    private AddressDTO address;
}
